### DISCUSSION TITLE
Avoiding cascade on Dependent fields

### DISCUSSION THREAD

**Discussion by andypooletrioteca** [laravel/nova-issues#5972] [Ideas & Feature Requests]
Hi, it would be really appreciated to have a `->avoidCascade()` option for dependent fields where you could determine if changes on the current field should trigger syncs on their dependent fields. For example, I have a complex form where i have several fields depending on each other. I need them to trigger changes on each other, but this changes shouldn't trigger sync on the initial field to avoid infinite loops.

this could be achieved by adding a `$avoidCascade` property to the `SupportsDependentFields` trait

```
public function avoidCascade()
{
    $this->withMeta(['avoidCascade' => true]);
}
```

Adding an `avoidCascade` parameter to the `emitFieldValueChangeFunction` in the `FormEvents.js` mixin:

```
    emitFieldValueChange(attribute, value, avoidCascade = false) {
      Nova.$emit(`${attribute}-change`, value, avoidCascade)

      if (this.hasFormUniqueId === true) {
        Nova.$emit(`${this.formUniqueId}-${attribute}-change`, value, avoidCascade)
      }
    },
```

Adding the parameter to the emission of the event in the `DependentFormField.js` mixin:

```
    emitOnSyncedFieldValueChange() {
      return this.emitFieldValueChange(this.field.attribute, this.currentField.value, this.field.avoidCascade);
    },
```

and finally applying the `avoidCascade` option to the `dependentFormField` handler in the mounted function of the same file.

```
        this.watchedEvents[dependsOn] = (value, avoidCascade) => {
          this.watchedFields[dependsOn] = value

          this.dependentFieldDebouncer(() => {
            this.watchedFields[dependsOn] = value

            if (avoidCascade) {
              return;
            }

            this.syncField()
          })
        }
```

If this is implemented by Nova i can use any kind of field. otherwise i need to create custom fields for each type of dependent field where i need to avoid cascading.

**Comment by davidhemphill** [comment]
Can you post an example of a real-life situation where you would use this?

**Comment by andypooletrioteca** [comment]
I would need this in a very complex situation and i don't want to bother you with all the details, so i'll try to reduce it to a simple case:

imagine i have a form with three fields:
- product price
- extras price
- total price

they are related based on this simple equation:
product price + extras price = total price

if you change the product price, the total price should change
if you change the extras price, the total price should change
if you change the total price, the extras price should change

this would help an agent figure out how many extras a client could add to a purchase in case the client reveals his budget...

**Comment by crynobone** [comment]
Released with Laravel Nova 5.0.0

Feel free to open up a new issue if you're still experiencing this problem on the latest version.
